src/plugins/mission_server/include/plugins/mission_server/mission_server.h

```diff
@@ -0,0 +1,235 @@
+// WARNING: THIS FILE IS AUTOGENERATED! As such, it should not be edited.
+// Edits need to be made to the proto files
+// (see
+// https://github.com/mavlink/MAVSDK-Proto/blob/main/protos/mission_server/mission_server.proto)
+
+#pragma once
+
+#include <array>
+#include <cmath>
+#include <functional>
+#include <limits>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "plugin_base.h"
+
+namespace mavsdk {
+
+class System;
+class MissionServerImpl;
+
+/**
+ * @brief Enable waypoint missions.
+ */
+class MissionServer : public PluginBase {
+public:
+    /**
+     * @brief Constructor. Creates the plugin for a specific System.
+     *
+     * The plugin is typically created as shown below:
+     *
+     *     ```cpp
+     *     auto mission_server = MissionServer(system);
+     *     ```
+     *
+     * @param system The specific system associated with this plugin.
+     */
+    explicit MissionServer(System& system); // deprecated
+
+    /**
+     * @brief Constructor. Creates the plugin for a specific System.
+     *
+     * The plugin is typically created as shown below:
+     *
+     *     ```cpp
+     *     auto mission_server = MissionServer(system);
+     *     ```
+     *
+     * @param system The specific system associated with this plugin.
+     */
+    explicit MissionServer(std::shared_ptr<System> system); // new
+
+    /**
+     * @brief Destructor (internal use only).
+     */
+    ~MissionServer();
+
+    /**
+     * @brief Type representing a mission item.
+     *
+     * A MissionItem can contain a position and/or actions.
+     * Mission items are building blocks to assemble a mission,
+     * which can be sent to (or received from) a system.
+     * They cannot be used independently.
+     */
+    struct MissionItem {
```

Comment:
Ok, so you decided that you want to implement/support the Mission API rather than the full MissionRaw API, right?

Reply:
For now, perhaps it makes more sense to use raw types